### PR TITLE
Improve documentation for workers/base.py

### PR DESCRIPTION
**Pull request by mari-linhares**

### CODE REVIEW

File: syft/workers/base.py [hunk 1]
```diff
@@ -583,7 +599,7 @@ def get_worker(self, id_or_worker, fail_hard=False):
 
         return id_or_worker
 
-    def add_worker(self, worker):
+    def add_worker(self, worker: "BaseWorker"):
```
Comment (robert-wagner) [inline syft/workers/base.py]: Why is this a string?

File: syft/workers/base.py [hunk 2]
```diff
@@ -348,12 +355,20 @@ def execute_command(self, message):
                 )
                 raise ResponseSignatureError(return_ids.generated)
 
-    def send_command(self, recipient, message, return_ids=None):
+    def send_command(
+        self, recipient: "BaseWorker", message: str, return_ids: str = None
```
Comment (robert-wagner) [inline syft/workers/base.py]: Why is base worker a string

File: syft/workers/base.py [hunk 3]
```diff
@@ -59,7 +59,13 @@ class BaseWorker(AbstractWorker):
     """
 
     def __init__(
-        self, hook, id=0, data=None, is_client_worker=False, log_msgs=False, verbose=False
+        self,
+        hook: "sy.TorchHook",
```
Comment (robert-wagner) [inline syft/workers/base.py]: Why is hook a string

File: syft/workers/base.py [hunk 4]
```diff
@@ -630,7 +646,7 @@ def add_worker(self, worker):
             )
         self._known_workers[worker.id] = worker
 
-    def add_workers(self, workers):
+    def add_workers(self, workers: List["BaseWorker"]):
```
Comment (robert-wagner) [inline syft/workers/base.py]: Why is base worker a string

File: syft/workers/base.py (hunk 1 above)
Comment (LaRiffle) [reply syft/workers/base.py]: You can mention a type if you are in its own class, that's kind the way of handling it

Comment (robert-wagner) [reply syft/workers/base.py]: Ah okay

File: syft/workers/base.py (hunk 3 above)
Comment (mari-linhares) [reply syft/workers/base.py]: TorchHook imports BaseWorker. So I can't import hook on BaseWorker. The way to handle this is using [Forward references](https://www.python.org/dev/peps/pep-0484/#forward-references)

**Pull request merged**